src/common/hooks/account/use-get-account-nonce.ts

```diff
@@ -0,0 +1,20 @@
+import { apiClientState } from '@store/common/api-clients';
+import { currentAccountStxAddressState } from '@store/accounts';
+import { useQuery, UseQueryOptions } from 'react-query';
+import { useAtomValue } from 'jotai/utils';
+import { currentNetworkState } from '@store/network/networks';
+
+export function useGetAccountNonce(reactQueryOptions: UseQueryOptions = {}) {
+  const principal = useAtomValue(currentAccountStxAddressState);
+  const network = useAtomValue(currentNetworkState);
+  const { accountsApi } = useAtomValue(apiClientState);
+  const fetchNonce = () => {
+    if (!principal) return;
+    return accountsApi.getAccountNonces({ principal });
+  };
+
+  return useQuery(['getAccountNonce', principal, network], fetchNonce, {
```

Comment:
Why name the key `get`? Couldn't it just be `AccountNonce`? We should have a naming convention for keys imo.

Reply:
Agree, I will remove the get. I think the key should start with a lowercase and it shall describe the result.

Reply:
I agree, good call. 👍